internal/validations/clusterlogforwarder/validate_clusterlogforwarderspec_test.go

```diff
@@ -986,6 +986,125 @@ var _ = Describe("Validate clusterlogforwarderspec", func() {
 			conds := clfStatus.Pipelines["someDefinedPipeline"]
 			Expect(conds).To(HaveCondition(loggingv1.ValidationCondition, true, loggingv1.ValidationFailureReason, "invalid: custom ClusterLogForwarders cannot forward to the `default` log store, unrecognized outputs: .?default.?"))
 		})
+
+		DescribeTable("gcl and prune filter", func(pruneSpec loggingv1.PruneFilterSpec) {
+			gclOutput := loggingv1.OutputSpec{
+
+				Name: "gcl-out",
+				Type: "googleCloudLogging",
+				OutputTypeSpec: loggingv1.OutputTypeSpec{
+					GoogleCloudLogging: &loggingv1.GoogleCloudLogging{BillingAccountID: "billingAccountID"},
+				},
+			}
+
+			pruneHost := loggingv1.FilterSpec{
+				Name: "prune",
+				Type: loggingv1.FilterPrune,
+				FilterTypeSpec: loggingv1.FilterTypeSpec{
+					PruneFilterSpec: &pruneSpec,
+				},
+			}
+
+			forwarderSpec.Outputs = []loggingv1.OutputSpec{gclOutput}
+			forwarderSpec.Filters = []loggingv1.FilterSpec{pruneHost}
+
+			forwarderSpec.Pipelines = []loggingv1.PipelineSpec{
+				{
+					Name:       "gclPruneHost",
+					OutputRefs: []string{"gcl-out"},
+					InputRefs:  []string{loggingv1.InputNameApplication},
+					FilterRefs: []string{"prune"},
+				},
+			}
+			clfStatus.Outputs.Set("gcl-out", condReady)
+
+			verifyPipelines("custom-clf-name", &forwarderSpec, clfStatus)
+			Expect(clfStatus.Pipelines).To(HaveLen(1))
+			Expect(clfStatus.Pipelines).To(HaveKey("gclPruneHost"))
+			conds := clfStatus.Pipelines["gclPruneHost"]
+			Expect(conds).To(HaveCondition(loggingv1.ConditionReady, false, loggingv1.ReasonInvalid, "googleCloudLogging cannot prune `.hostname` field.+"))
```

Comment:
Shouln't this HaveCondition of ConditionTrue ? since the condition exists (i.e. we have a failure)?  There was a bug in or condition creation that was not seting "ConditionTrue" correctly but my belief was that was since corrected

Reply:
The condition here is `Type: ConditionReady` so I believe the condition should be false since it should be failing validation and thus is not in a ready state. If it was a `ValidationCondition` it would be `True`